hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -70,120 +63,61 @@
-				txnCtx.setStatus(validity);
-				return;
-			}
+		/* --- Translate from gRPC types --- */
+		final var txnBody = txnCtx.accessor().getTxn();
+		final var op = txnBody.getCryptoUpdateAccount();
 
-			ledger.customize(target, customizer);
-			txnCtx.setStatus(SUCCESS);
-		} catch (MissingAccountException mae) {
-			txnCtx.setStatus(INVALID_ACCOUNT_ID);
-		} catch (DeletedAccountException aide) {
-			txnCtx.setStatus(ACCOUNT_DELETED);
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
-		}
-	}
-
-	private ResponseCodeEnum sanityCheck(AccountID target, HederaAccountCustomizer customizer) {
-		if (!ledger.exists(target) || ledger.isSmartContract(target)) {
-			return INVALID_ACCOUNT_ID;
-		}
+		/* --- Load the model objects --- */
+		final var accountId = Id.fromGrpcAccount(op.getAccountIDToUpdate());
+		final var accountToBeUpdated = accountStore.loadAccount(accountId);
 
-		final var changes = customizer.getChanges();
-		final var keyChanges = customizer.getChanges().keySet();
+		validateFalse(accountToBeUpdated.isSmartContract(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 
-		if (ledger.isDetached(target)) {
-			if (!keyChanges.equals(EXPIRY_ONLY)) {
-				return ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
-			}
+		if (op.hasExpirationTime()) {
+			validateFalse(accountToBeUpdated.getExpiry() > op.getExpirationTime().getSeconds(), EXPIRATION_REDUCTION_NOT_ALLOWED);
+			validateFalse(!validator.isValidExpiry(op.getExpirationTime()), INVALID_EXPIRATION_TIME);
 		}
 
-		if (keyChanges.contains(AccountProperty.EXPIRY)) {
-			final long newExpiry = (long)changes.get(AccountProperty.EXPIRY);
-			if (newExpiry < ledger.expiry(target)) {
-				return EXPIRATION_REDUCTION_NOT_ALLOWED;
-			}
+		if (op.hasMaxAutomaticTokenAssociations()) {
+			validateFalse(op.getMaxAutomaticTokenAssociations().getValue() < accountToBeUpdated.getAlreadyUsedAutomaticAssociations(), EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT);
+			//TODO dynamicProperties.maxTokensPerAccount() returns inconsistent values
+			validateFalse(op.getMaxAutomaticTokenAssociations().getValue() > dynamicProperties.maxTokensPerAccount(), REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
 		}
 
-		if (keyChanges.contains(AccountProperty.MAX_AUTOMATIC_ASSOCIATIONS)) {
-			final long newMax = (int) changes.get(AccountProperty.MAX_AUTOMATIC_ASSOCIATIONS);
-			if (newMax < ledger.alreadyUsedAutomaticAssociations(target)) {
-				return EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT;
-			}
-			if (newMax > dynamicProperties.maxTokensPerAccount()) {
-				return REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT;
-			}
-		}
+		/* --- Do the business logic --- */
+		Account.updateFromGrpc(accountToBeUpdated, op, txnCtx.consensusTime().getEpochSecond());
 
-		return OK;
-	}
+		/* --- Persist the models --- */
+		this.accountStore.persistAccount(accountToBeUpdated);
```

Comment:
nit: `this `keyword in this particular context seems redundant, same on line 119

Reply:
agreed and removed